src/plugins/dashboard/public/dashboard_api/unsaved_changes.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { BehaviorSubject } from 'rxjs';
+import type { DashboardContainerInput } from '../../common';
+
+export function initializeUnsavedChanges(
+  anyMigrationRun: boolean,
+  lastSavedInput: DashboardContainerInput
+) {
+  const hasRunMigrations$ = new BehaviorSubject(anyMigrationRun);
```

Comment:
Does this need to be a behaviour subject? Does this value ever change after fetching the dashboard SO? 🤔

Reply:
`hasRunMigrations$` is updated on save when `setLastSavedInput` is called.